src/ducts/CCPACSDuctAssembly.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+* Copyright (C) 2007-2022 German Aerospace Center (DLR/SC)
+*
+* Created: 2022-03-24 Jan Kleinert <Jan.Kleinert@dlr.de>
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+/**
+* @file
+* @brief  Implementation of CPACS duct handling routines.
+*/
+
+//#define USE_TIGL_FUSER
+
+#include "CCPACSDuctAssembly.h"
+#include "CCPACSDuct.h"
+#include "Debugging.h"
+#include "CCutShape.h"
+#include "CNamedShape.h"
+#include "CTiglError.h"
+
+#ifdef USE_TIGL_FUSER
+#include "CFuseShapes.h"
+#else
+#include "TopTools_ListOfShape.hxx"
+#include "BRepAlgoAPI_Fuse.hxx"
+#include "CBooleanOperTools.h"
+#endif
+
+
+
+namespace tigl {
+
+CCPACSDuctAssembly::CCPACSDuctAssembly(CCPACSDucts* parent, CTiglUIDManager* uidMgr)
+  : generated::CPACSDuctAssembly(parent, uidMgr)
+  , CTiglRelativelyPositionedComponent(&m_parentUID, &m_transformation)
+{
+    for (auto const& uid: m_ductUIDs.GetUIDs()) {
+        auto& duct = GetParent()->GetDuct(uid);
+        duct.RegisterInvalidationCallback([&](){ this->Invalidate(); });
+    }
+}
+
+std::string CCPACSDuctAssembly::GetDefaultedUID() const
+{
+    return generated::CPACSDuctAssembly::GetUID();
+}
+
+TiglGeometricComponentType CCPACSDuctAssembly::GetComponentType() const
+{
+    return TIGL_COMPONENT_DUCT;
+}
+
+TiglGeometricComponentIntent CCPACSDuctAssembly::GetComponentIntent() const
+{
+    // needs to be physical, so that transformation relative to parent works
+    return TIGL_INTENT_PHYSICAL;
+}
+
+void CCPACSDuctAssembly::InvalidateImpl(const boost::optional<std::string>&) const
+{
+    CTiglAbstractGeometricComponent::Reset();
+}
+
+PNamedShape CCPACSDuctAssembly::BuildLoft() const
+{
+    auto const& trafo = GetTransformationMatrix();
+    if (m_ductUIDs.GetUIDs().size() == 0) {
+        return PNamedShape();
```

Comment:
This special case is not covered by any tests currently (https://app.codecov.io/gh/DLR-SC/tigl/compare/889/diff). I am undecided. Should we add a unit test?

Reply:
I think it would be a good idea to add such a unit test. I can adapt the testDuct.cpp and simple_duct.xml files to cover this case.